pkg/nodebootstrap/managed_al2.go

```diff
@@ -82,12 +81,29 @@ func (m *ManagedAL2) UserData() (string, error) {
 	return base64.StdEncoding.EncodeToString(buf.Bytes()), nil
 }
 
-func makeCustomAMIUserData(ng *api.NodeGroupBase) (string, error) {
+func makeCustomAMIUserData(ng *api.NodeGroupBase, mimeBoundary string) (string, error) {
+	var (
+		buf     bytes.Buffer
+		scripts []string
+	)
+
+	if len(ng.PreBootstrapCommands) > 0 {
+		scripts = append(scripts, ng.PreBootstrapCommands...)
+	}
+
 	if ng.OverrideBootstrapCommand != nil {
-		return base64.StdEncoding.EncodeToString([]byte(*ng.OverrideBootstrapCommand)), nil
+		scripts = append(scripts, *ng.OverrideBootstrapCommand)
 	}
 
-	return "", nil
+	if len(scripts) == 0 {
+		return "", nil
+	}
+
+	if err := createMimeMessage(&buf, scripts, nil, mimeBoundary); err != nil {
```

Comment:
@Callisto13 or Any one who is going to review this
Would it be better to use cloudConfig instead of MIME format?

cloudConfig would allow all the preBootstrapCommands to be run as single script file. Where as with MIME format each string in preBootstrapCommands is it's script file.

The benefit of allowing all the preBootstrapCommands to run as one script file is, any environment variables initialized earlier will be applicable to later commands.

Other alternative to make preBootstrapCommands as one script while using MIME format is, I can join the preBootstrapCommands with a newline char, like below

```

	if len(ng.PreBootstrapCommands) > 0 {
		scripts = append(scripts, ng.PreBootstrapCommands...)
	}

```

please share your thoughts.

Reply:
That makes sense to me @kishoregv; we don't appear to add the Override commands in the MIME format when using custom AMIs, so if adding the Pre- commands as a single cloudconfig script works, then that should be fine.

I may be missing some context here, however; @cPu1 do you know if there is a good reason to use MIME in this case?

Reply:
@Callisto13 If the PR looks good you folks, I am ok to merge the PR as is without any future changes that I was proposing above.

Thanks

Reply:
👍  sounds good

If you were able to manually verify that the current changes fix the bug, then I am happy with it; the priority is to get the bug fixed.

If you'd like to submit a follow-up pr to do the suggested change when we have more time to look into it, please do.

Reply:
I have manually verified that the current changes fix the bug.